Test/FMI3/fmi3_basic_capi_test.cpp

```diff
@@ -0,0 +1,524 @@
+        REQUIRE( status == 0 );
+        fmi3_string_t instanceName = "Test ME model instance";
+        fmi3_string_t fmuInstantiationToken;
+        fmi3_string_t resourcePath = "";
+        fmi3_boolean_t visible = fmi3_false;
+        fmi3_boolean_t loggingOn = fmi3_false;
+        dummy_fmi3_instance_environment_t instance_env = {};
+        jm_status_enu_t instantiate_status = fmi3_import_instantiate_model_exchange(
+            fmu,
+            instanceName,
+            resourcePath,
+            visible,
+            loggingOn,
+            ((fmi3_instance_environment_t)&instance_env),
+            (fmi3_log_message_callback_ft)dummy_log_message_callback
+        );
+        REQUIRE( instantiate_status == jm_status_success );
+        fmi3_status_t status_from_float_set = fmi3_import_set_float64(
+            fmu,
+            value_references,
+            n_value_references,
+            values,
+            n_values
+        );
+        REQUIRE( status_from_float_set == fmi3_status_ok );
+
+        fmi3_float64_t values_from_get[n_values] = {0.0, 0.0, 0.0, 0.0};
+        fmi3_status_t status_from_float_get = fmi3_import_get_float64(
+            fmu,
+            value_references,
+            n_value_references,
+            values_from_get,
+            n_values
+        );
+        REQUIRE( status_from_float_get == fmi3_status_ok );
+        // We use indivdual calls to REQUIRE to make a potential error point to the exact line
+        REQUIRE( values[0] == values_from_get[0] );
+        REQUIRE( values[1] == values_from_get[1] );
+        REQUIRE( values[2] == values_from_get[2] );
+        REQUIRE( values[3] == values_from_get[3] );
+    }
+
+    SECTION( "Test instantiate and set values on int64 array" ) {
+        size_t n_value_references = 1;
+        /* This value reference does not matter because the Dummy FMU
+            implementation for set/get int64 uses a fixed dummy array.
+        */
+        const fmi3_value_reference_t value_references[n_value_references] = {1};
+        size_t n_int_values = 2;
+        const fmi3_int64_t int64_values[n_int_values] = {-9223372036854775800, 9223372036854775800};
+        jm_status_enu_t status = fmi3_import_create_dllfmu(fmu, fmi3_fmu_kind_me, &inst_env, fmi3_dummy_log_callback);
+        REQUIRE( status == 0 );
+        fmi3_string_t instanceName = "Test ME model instance";
+        fmi3_string_t fmuInstantiationToken;
+        fmi3_string_t resourcePath = "";
+        fmi3_boolean_t visible = fmi3_false;
+        fmi3_boolean_t loggingOn = fmi3_false;
+        dummy_fmi3_instance_environment_t instance_env = {};
+        jm_status_enu_t instantiate_status = fmi3_import_instantiate_model_exchange(
+            fmu,
+            instanceName,
+            resourcePath,
+            visible,
+            loggingOn,
+            ((fmi3_instance_environment_t)&instance_env),
+            (fmi3_log_message_callback_ft)dummy_log_message_callback
+        );
```

Comment:
You can move the whole dll and fmu instantiation up one SECTION to avoid code duplication.